azure-kusto-data/azure/kusto/data/request.py

```diff
@@ -140,6 +167,24 @@ def application_key(self):
         """The application key to use for authentication when federated authentication is used"""
         return self._internal_dict.get(self.ValidKeywords.application_key)
 
+    @property
+    def application_certificate(self):
+        """A PEM encoded certificate private key."""
+        return self._internal_dict.get(self.ValidKeywords.application_certificate)
+
+    @application_certificate.setter
+    def application_certificate(self, value):
+        self[self.ValidKeywords.application_certificate] = value
+
+    @property
+    def application_certificate_thumbprint(self):
+        """hex encoded thumbprint of the certificate."""

```

Comment:
Thumbprint is always a hexadecimal string

Reply:
I took it from here:
https://github.com/AzureAD/azure-activedirectory-library-for-python/blob/ef3a8814540684b6fbd709a249dc7c522f94584a/adal/authentication_context.py#L245